src/domain/flow-system/services/src/flow/flow_service_impl.rs

```diff
@@ -1226,14 +1311,18 @@ impl FlowService for FlowServiceImpl {
         &self,
         flow_id: FlowID,
     ) -> Result<FlowState, CancelScheduledTasksError> {
-        let mut flow = Flow::load(flow_id, self.flow_event_store.as_ref()).await?;
+        let flow_event_store = self.catalog.get_one::<dyn FlowEventStore>().unwrap();
+        let task_scheduler = self.catalog.get_one::<dyn TaskScheduler>().unwrap();
```

Comment:
It's clearly starting to feel like we have 2 flow managers :-)
One is controlling transaction, another one is dependent on externally managed transactions.